### DISCUSSION TITLE
How to Install One Block [ DOCUMENTATION ]

### DISCUSSION THREAD

**Discussion by josephdpurcell** [itzg/docker-minecraft-bedrock-server#386] [Show and tell]
One of my missions with exploring this project was to see if I could figure out how to get a OneBlock variation to work. And I did! So, I wanted to share.

This is going to use https://www.vatonage.com/post/one-block for the mod. I am new to Minecraft and it's unclear to me which sites are reputable for downloading files from. At least for the purposes of OneBlock this mod is pretty amazing.

# Requirements

You are doing this guide with a fresh installation. I don't know if this works with multiple worlds in the same server.

This assumes you're using this project to run your Minecraft server.

It assumes you know how to read and edit JSON. If you do not check out a "Tutorial" here https://github.com/burningtree/awesome-json#tutorials.

# 1 - Download the mod

Go to https://www.vatonage.com/post/one-block and scroll down to the "Download Addon" button.

This will take you out to a MediaFire site where you'll download a file called `OneBlock.mcworld`.

When I downloaded it 2023-12-16 the sashum and md5sum (whichever you prefer) was:

```
$ md5sum OneBlock.mcworld 
1259b591211f22dc72922d852e6efe5a  OneBlock.mcworld

$ shasum OneBlock.mcworld 
8b54012a09e8dc02f3468720267ae5c551d38c57  OneBlock.mcworld
```

This file isn't versioned so I can't share what version it is. I can say if you unzip the files they were last modified 2023-01-08.

# 2 - Obtain the world name

With behavior packs and resource packs you have to find some detailed information before you can know where to move it.

But, this is a world file which ends in `.mcworld`. You can see this referenced on Microsoft's documentation here: https://learn.microsoft.com/en-us/minecraft/creator/documents/minecraftfileextensions?view=minecraft-bedrock-stable#mcworld, but it doesn't give a lot of info so you might want to read this instead: https://learn.microsoft.com/en-us/minecraft/creator/documents/createaworldtemplate?view=minecraft-bedrock-stable.

My understanding of these `mcworld` files is if you just took any folder in `data/worlds/` and zip'd it up and gave it a `.mcworld` extension you would have a valid `.mcworld` file, but I'm not knowledgeable on this topic.

You need the world name and I think it's just the name of the file? In this case it's `OneBlock`. But, I also see a `levelname.txt` and the documentation says that THIS is the level name. For the sake of this document I'm going to say the name is `OneBlock` because that's what worked for me.

# 3 - Copy the mod file

First, your Minecraft server should have a data directory somewhere, it depends on how you set it up. In my case I mounted the local directory, you can see my docker-compose.yml in https://github.com/itzg/docker-minecraft-bedrock-server/issues/194#issuecomment-1858783824.

The file we're going to copy there is the `.mcworld` file we downloaded. Here's how that looks:

```
$ mkdir ./data/worlds/OneBlock/

# Note: we rename the file from mcworld extension to a zip to ensure our unzip tool knows its a zip
$ cp ~/Downloads/OneBlock.mcworld ./data/worlds/OneBlock/OneBlock.zip

$ cd ./data/worlds/OneBlock/

$ unzip OneBlock.zip

$ ls ./data/worlds/OneBlock/
behavior_packs  level.dat      levelname.txt   world_behavior_pack_history.json  world_icon.jpeg                   world_resource_packs.json
db              level.dat_old  resource_packs  world_behavior_packs.json         world_resource_pack_history.json
```

You can see that the contents of the `OneBlock.mcworld` file are all inside the `data/worlds/OneBlock` directory. And, also worth pointing out that it includes its own `behavior_packs` and `resource_packs` -- so these `.mcworld` files have a lot in them.

# 4 - Set the world name

Use the `LEVEL_NAME` environment variable to set your world name to `OneBlock`. If you're using docker compose you can do it like this:

```yaml
version: '3'
services:
  minecraft:
    environment:
      LEVEL_NAME: 'OneBlock'
```

I'm omitting lots of lines here, but refer to my [docker-compose.yml](https://github.com/itzg/docker-minecraft-bedrock-server/issues/194#issuecomment-1858783824) for details.

# 5 - Restart the server and connect

Restart your server. In the case of using docker compose you would run:

```
docker compose stop
```

Then:

```
docker compose up
```

On startup you'll see various mods load. When your Minecraft client connects you will be prompted to "Download & Join" which will install various resource packs.

You will spawn on top of a single block.

**Comment by josephdpurcell** [comment]
I did encounter at least two issues with the OneBlock.mcworld I downloaded:

1. The server logs show lots of errors https://gist.github.com/josephdpurcell/49168e820f3663abd1c3a6828548f86e, and during gameplay I did see a couple blocks with IDs instead of labels.
2. Mobs don't spawn.

The mobs not spawning was a bummer, but I do see someone else encountered the same issue: https://www.vatonage.com/post/one-block?commentId=b345723f-4fb9-460e-a34d-922868c6d830.

I checked and the mob spawning game rule is enabled by default and fiddling with it didn't seem to affect anything.

**Comment by josephdpurcell** [comment]
I wonder if mobs not spawning is because I need to make more blocks:

> There must be no players or the world spawn point within a 24 radius block distance (spherical) of the spawning block

Source: https://minecraft.wiki/w/Spawn

**Comment by josephdpurcell** [comment]
Confirmed: the mob spawn happens if you build the world large enough.

**Comment by itzg** [comment]
Do you mind if I convert these "documentation issues" into discussion items, such as in Show & Tell section? Otherwise, the stale issue bot will eventually decide to close these.

**Comment by josephdpurcell** [comment]
That's a great idea, thank you.

I see now that Discussions have categories (Discussions is still a new feature to me). That seems like a good place.

I don't know when I'll be meandering through here next but if I see these tickets haven't moved I will do so.

Thanks for this excellent project! It's very fun.
